src/main/scala/cromwell/Main.scala

```diff
@@ -142,17 +149,16 @@ class Main private[cromwell](enableSysExit: Boolean, managerSystem: () => Workfl
     } yield Paths.get(args.head).resolveSibling(path)
   }
 
-  private[this] def runWorkflow(workflowSourceFiles: WorkflowSourceFiles, metadataPath: Option[Path]): Int = {
+  private[this] def runWorkflow(workflowSourceFiles: WorkflowSourceFiles, metadataPath: Option[Path])(implicit timeout: Timeout): Int = {
     val workflowManagerSystem = managerSystem()
-    val singleWorkflowRunner = SingleWorkflowRunnerActor.props(workflowSourceFiles, metadataPath,
+    val runnerProps = SingleWorkflowRunnerActor.props(workflowSourceFiles, metadataPath,
       workflowManagerSystem.workflowManagerActor)
-    workflowManagerSystem.actorSystem.actorOf(singleWorkflowRunner, "SingleWorkflowRunnerActor")
-    workflowManagerSystem.actorSystem.awaitTermination() // And now we just wait for the magic to happen
-    /*
-     * NOTE: When the actor system is shutdown, no communication of success or failure is messaged or recorded
-     * hence the exit code will always be zero.
-     */
-    exit(0)
+    val runner = workflowManagerSystem.actorSystem.actorOf(runnerProps, "SingleWorkflowRunnerActor")
+
+    val futureResult = runner ? RunWorkflow
+    Await.ready(futureResult, Duration.Inf)
```

Comment:
They are evil incarnate in an actor, but happily this is not an actor and needs to wait on the `SingleWorkflowRunnerActor` to finish running the workflow.